src/UI/Actions.hs

```diff
@@ -697,14 +700,25 @@ listSetSelectionEnd
   => Lens' s (L.GenericList Name t a) -> s -> s
 listSetSelectionEnd l = over l (L.listMoveTo (-1))
 
-applySearch :: AppState -> T.EventM Name AppState
-applySearch s = runExceptT (Notmuch.getThreads searchterms (view (asConfig . confNotmuch) s))
-                >>= pure . ($ s) . either setError updateList
+applySearch :: (MonadIO m) => AppState -> m AppState
+applySearch s = do
+  r <- runExceptT (Notmuch.getThreads searchterms (view (asConfig . confNotmuch) s))
+  case r of
+    Left e -> pure $ setError e s
+    Right threads -> notifyNumThreads s threads $> updateList threads s
    where searchterms = currentLine $ view (asMailIndex . miSearchThreadsEditor . E.editContentsL) s
          updateList vec =
            over (asMailIndex . miThreads . listList) (L.listReplace vec (Just 0))
            . set (asMailIndex . miThreads . listLength) Nothing
 
+-- | Fork a thread to compute the length of the container and send a
+-- NotifyNumThreads event.  'seq' ensures that the work is actually
+-- done by the spawned thread.
+notifyNumThreads :: (MonadIO m, Foldable t) => AppState -> t a -> m ()
```

Comment:
Would we gain better control/robustness if we'd use the async package? Guess the current use doesn't strike me as too complex, so perhaps it's something for later.